pkg/util/deployment/deployment.go

```diff
@@ -270,6 +279,16 @@ func waitForReplicaSetUpdated(c clientset.Interface, desiredGeneration int64, na
 	})
 }
 
+func waitForPodsHashPopulated(c clientset.Interface, namespace, name string) error {
+	return wait.Poll(1*time.Second, 1*time.Minute, func() (bool, error) {
+		rs, err := c.Extensions().ReplicaSets(namespace).Get(name)
+		if err != nil {
+			return false, err
+		}
+		return rs.Status.FullyLabeledReplicas == rs.Spec.Replicas, nil
```

Comment:
Recording in-person discussion: observedGeneration is checked in the caller currently, but I'm concerned about new calls being introduced later where that isn't true.